proposals/0245-sharing-wifi-ssid-and-password.md

```diff
@@ -18,627 +18,1205 @@ If WiFi credentials are automatically generated by the IVI system, the password
-    }
+}
 ```
-Permissions required to use above mentioned API's are as follows:
+
+__Note:__ Keep in mind, if `encryption_required` for `OnNetworkConfiguration` is set to `true`, Mobile application will not receive notification until secure service is not established.
+
+## Troubleshooting and race conditions
+
+### Race condition in case of multiple device connection
+
+#### Registration race
+
+From the Core perspective registration process is sequential and since it will cache registration order, there gonna be no race condition during registration.
+
+#### Capability race
+
+##### Vehicle is the hotspot
+
+After the application from the device is connected and Vehicle should be hotspot, Core will send `OnNetworkConfiguration` notification to the application and to HMI. Since it is Vehicle responsibility to start Hotspot, Core should wait for `OnCapabilityUpdate` with the updated capabilities. Only after this notification, Core is ready to process feature enabled application from another device.
+
+#### Mobile device is the hotspot
+
+After Core has decided, that Mobile device should be an Access point, Core will send `OnNetworkConfiguration` notification to the application and HMI. Since HMI has nothing to do before establishing WiFi connection. So the HMI capability will be updated only after connection attempt to the Mobile hotspot. Before this point, Core will not be able to process feature enabled applications from other devices.
+
+### WiFi disconnection
+
+Core knows nothing of WiFi connection state. The application which uses this feature, will connect via secondary transport. After the connection is established, Core will not proceed feature RPCs to the application. Once WiFi connetion lost, secondary transport connection will be broken, so the feature will be enabled again for the Application until TCP connection will be established.
+
+### Manual WiFi connection
+
+Mobile device is still can be connected to the Vehicle manually (or vice versa). In this case, if an application is connected via other transports, it will receive TransportEventUpdate and applications will be able to connect via WiFi, so the feature is not requred after this point. Core should not process `OnNetworkConfiguration` and `JoinNetwork` for the application, which is already connected via WiFi.
+
+However, if there are no applications connected via WiFi, Core knows nothing of the established connection, so the first feature enable application will still be able to use the feature.
+
+![WiFi AP created manually. Disconnected](../assets/proposals/0245-sharing-wifi-ssid-and-password/WiFi_AP_created_manually_Disconnected.png)
+
+![Negative cases](../assets/proposals/0245-sharing-wifi-ssid-and-password/negative_case.png)
+
+## Potential Downsides
+
+SDL is not able to check that 2 different transport belongs to the same device.
+On device connected by 2 different devices will be 2 devices for SDL. 

```

Comment:
I'm not sure what you mean here.  Can you please rephrase?

Reply:
Fixed in 00d9e1cf4edbbef38be65f55dceb1e577db984fd

Reply:
I am not seeing these changes reflected.  Can you please update?

Reply:
Changed to `The device connected by 2 different types of transport SDL will understand as different 2 devices.`

Reply:
I've left a new comment with feedback.